addons/mail/models/mail_followers.py

```diff
@@ -116,39 +116,57 @@ def _get_recipient_data(self, records, subtype_id, pids=None, cids=None):
           user groups of partner (void as irrelevant if channel ID),
         """
         if records and subtype_id:
+            query_partners = """
+    SELECT partner.id as pid, cast(NULL as INTEGER) AS cid,
+            partner.active as active, partner.partner_share as pshare, NULL as ctype,
+            users.notification_type AS notif, array_agg(groups.id) AS groups
+        FROM res_partner partner
+        LEFT JOIN res_users users ON users.partner_id = partner.id AND users.active
```

Comment:
partner_id is required on res.users model, so a JOIN is smarter than a LEFT JOIN, even if it doesn't change query performance, AFAIK.